File: 0401-0500/0435-Non-overlapping Intervals/0435-Non-overlapping Intervals.py
import math
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        count = 0
        prev = -math.inf
        for interval in intervals:
            if interval[0] < prev:
                count += 1
                prev = min(prev, interval[1])
            else:
                prev = interval[1]
        return count

File: 0401-0500/0435-Non-overlapping Intervals/0435-Non-overlapping Intervals.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        int n = intervals.size();
        if (n == 0) {
            return 0;
        }
        sort(intervals.begin(), intervals.end());
        int count = 0, prev = intervals[0][1];
        for (int i = 1; i < n; ++i) {
            if (intervals[i][0] < prev) {
                ++count;
                prev = min(prev, intervals[i][1]);
            }
            else {
                prev = intervals[i][1];
            }
        }
        
        return count;
    }
};
